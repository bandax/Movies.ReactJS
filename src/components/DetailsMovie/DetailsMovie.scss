.details-movie {
  text-align: center;
}
