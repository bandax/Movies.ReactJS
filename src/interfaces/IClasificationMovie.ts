export interface IClasification {
    id: string;
    name: string;
}
