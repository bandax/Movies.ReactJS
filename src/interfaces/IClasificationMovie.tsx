export interface IClasification {
    id: string,
    name: string
}

export interface IClasificationMovieProps {
    clasificationTypes: IClasification[],    
}

