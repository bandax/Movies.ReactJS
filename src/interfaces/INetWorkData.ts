export interface INetworkData {
  url: string;
  body: string;
  contentType: string;
}
