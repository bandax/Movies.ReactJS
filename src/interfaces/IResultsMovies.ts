export interface IMovie {
    id: string;
    title: string;
    image: string;
    year: number;
    categories: ICategory[];
    releaseDate: string;
    review: string;
    url: string;    
    runtime: string;
}

export interface ICategory {
    id: string; 
    name: string;
}
