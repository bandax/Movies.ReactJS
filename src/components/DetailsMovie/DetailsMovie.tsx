import * as React from "react"
import "./DetailsMovie.scss"
import { IMovie } from "../../interfaces/IResultsMovies"

export interface IDetailsMovieProps {
    movie: IMovie
}

const DetailsMovie: React.FunctionComponent<IDetailsMovieProps> = (
    props: IDetailsMovieProps
) => {
    const movie = props.movie

    return (
        <div className="details-movie" key={movie.id}>
            <img className="poster-movie" src={movie.image} />
            <br />
            <span className="title-movie">{movie.title}</span>
            <span className="year-movie">{movie.year}</span>
            <span className="categories-movie">
                {movie.categories.map((category) => category.name).join(", ")}
            </span>
        </div>
    )
}

export { DetailsMovie }
