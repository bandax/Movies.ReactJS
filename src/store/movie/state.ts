import { IMovie } from "../../interfaces/IResultsMovies";

export interface MovieState {
  movies: IMovie[];
  loading: boolean;
  errorMessage: string;
}
