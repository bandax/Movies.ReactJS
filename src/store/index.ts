import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { movieReducer } from './movie/reducers';

const rootReducer = combineReducers({
  movie: movieReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(rootReducer, applyMiddleware(thunk));
