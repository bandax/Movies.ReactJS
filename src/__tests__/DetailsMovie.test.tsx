import * as React from 'react';
import { DetailsMovie } from '../components/DetailsMovie/DetailsMovie';
import thunk, { ThunkDispatch } from 'redux-thunk';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import configureMockStore, { MockStoreCreator } from 'redux-mock-store';
import { MovieState } from '../store/movie/state';
import { Action } from 'redux';
import { RootState } from '../store/index';
import clasificationTypes from '../data/clasifications.json';
import { IMovieData } from '../interfaces/IMovieData';

type DispatchExts = ThunkDispatch<MovieState, void, Action>;
const middleware = [thunk];
const state: RootState = {
  movie: {
    movie: null,
    errorMessage: '',
    findMovies: true,
    loading: false,
    movies: [],
  },
};

const movie: IMovieData = {
  id: 0,
  title: 'Avengers Infinity War',
  poster_path:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png',
  release_date: new Date().toLocaleDateString(),
  budget: 123,
  revenue: 123,
  vote_average: 123,
  vote_count: 123,
  runtime: 412,
  tagline: '12343',
  overview: 'tests',
  genres: ['Action & Adventure', 'Comedy'],
};

describe('<AddMovie />', () => {
  let onEditMovie;
  let onDeleteMovie;

  beforeEach(() => {
    onEditMovie = jest.fn();
    onDeleteMovie = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render a Details Movie Component', async () => {
    const { asFragment } = render(
      <DetailsMovie
        movie={movie}
        onEditMovie={onEditMovie}
        onDeleteMovie={onDeleteMovie}
      />,
    );

    const fragment = asFragment();

    expect(fragment).toMatchSnapshot();
  });

  it('should allow to call edit button', async () => {
    const { getByTestId, getByText } = render(
      <DetailsMovie
        movie={movie}
        onEditMovie={onEditMovie}
        onDeleteMovie={onDeleteMovie}
      />,
    );
    const button = getByText('Edit Movie');

    userEvent.click(button);

    await waitFor(() => {
      expect(onEditMovie).toHaveBeenCalled();
    });
  });

  it('should allow to call delete movie button', async () => {
    const { getByTestId, getByText } = render(
      <DetailsMovie
        movie={movie}
        onEditMovie={onEditMovie}
        onDeleteMovie={onDeleteMovie}
      />,
    );
    const button = getByText('Delete Movie');

    userEvent.click(button);

    await waitFor(() => {
      expect(onDeleteMovie).toHaveBeenCalled();
    });
  });
});
