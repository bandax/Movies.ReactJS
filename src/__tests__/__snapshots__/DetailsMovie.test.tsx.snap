// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddMovie /> should render a Details Movie Component 1`] = `
<DocumentFragment>
  <div
    class="details-movie"
  >
    <img
      class="poster-movie"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/Scarymovie4-logo.svg/1920px-Scarymovie4-logo.svg.png"
    />
    <br />
    <span
      class="title-movie"
    >
      Avengers Infinity War
    </span>
    <span
      class="year-movie"
    >
      2020
    </span>
    <span
      class="categories-movie"
    >
      Action & Adventure, Comedy
    </span>
    <button
      class=""
    >
      Edit Movie
    </button>
    <button
      class=""
    >
      Delete Movie
    </button>
    <a
      href="#/film/0"
    >
      Show Details Movie
    </a>
  </div>
</DocumentFragment>
`;
